pkg/analyze/velero.go

```diff
@@ -0,0 +1,697 @@
+		}
+
+		for _, restore := range restores {
+			if failedPhases[restore.Status.Phase] {
+				result := &AnalyzeResult{
+					Title: fmt.Sprintf("Restore %s", restore.Name),
+				}
+				result.IsFail = true
+				result.Message = fmt.Sprintf("Restore %s phase is %s", restore.Name, restore.Status.Phase)
+				results = append(results, result)
+				failures++
+			}
+		}
+		if failures == 0 {
+			results = append(results, &AnalyzeResult{
+				Title:   "Velero Restores",
+				IsPass:  true,
+				Message: fmt.Sprintf("Found %d restores", len(restores)),
+			})
+		}
+	}
+
+	return results
+}
+
+func analyzeSchedules(schedules []*velerov1.Schedule) []*AnalyzeResult {
+	results := []*AnalyzeResult{}
+	failures := 0
+	if len(schedules) > 0 {
+		for _, schedule := range schedules {
+			if schedule.Status.Phase == velerov1.SchedulePhaseFailedValidation {
+				result := &AnalyzeResult{
+					Title: fmt.Sprintf("Schedule %s", schedule.Name),
+				}
+				result.IsFail = true
+				result.Message = fmt.Sprintf("Schedule %s phase is %s", schedule.Name, schedule.Status.Phase)
+				results = append(results, result)
+				failures++
+			}
+		}
+		if failures == 0 {
+			results = append(results, &AnalyzeResult{
+				Title:   "Velero Schedules",
+				IsPass:  true,
+				Message: fmt.Sprintf("Found %d schedules", len(schedules)),
+			})
+		}
+	}
+	return results
+}
+
+func analyzeServerStatusRequests(serverStatusRequests []*velerov1.ServerStatusRequest) []*AnalyzeResult {
+	results := []*AnalyzeResult{}
+	// TODO
+	return results
+}
+
+func analyzeVolumeSnapshotLocations(volumeSnapshotLocations []*velerov1.VolumeSnapshotLocation) []*AnalyzeResult {
+	results := []*AnalyzeResult{}
+	failures := 0
+	if len(volumeSnapshotLocations) > 0 {
+		for _, volumeSnapshotLocation := range volumeSnapshotLocations {
+			if volumeSnapshotLocation.Status.Phase == velerov1.VolumeSnapshotLocationPhaseUnavailable {
+				result := &AnalyzeResult{
+					Title: fmt.Sprintf("Volume Snapshot Location %s", volumeSnapshotLocation.Name),
+				}
+				result.IsFail = true
+				// result.Strict = true
+				result.Message = fmt.Sprintf("Volume Snapshot Location %s phase is %s", volumeSnapshotLocation.Name, volumeSnapshotLocation.Status.Phase)
+				results = append(results, result)
+				failures++
+			}
+		}
+		if failures == 0 {
+			results = append(results, &AnalyzeResult{
+				Title:   "Velero Volume Snapshot Locations",
+				IsPass:  true,
+				Message: fmt.Sprintf("Found %d volume snapshot locations", len(volumeSnapshotLocations)),
+			})
+		}
+	}
+
+	return results
+}
+
+func analyzeLogs(logs map[string][]byte) []*AnalyzeResult {
+	results := []*AnalyzeResult{}
+	if len(logs) > 0 {
+		for _, logBytes := range logs {
+			logContent := string(logBytes)
+			result := &AnalyzeResult{
+				Title: fmt.Sprintf("Velero logs for pod [node-agent] "),
+			}
+			if strings.Contains(logContent, "permission denied") {
+				result.IsFail = true
```

Comment:
I'd leave this as a warning since logs in a support bundle may be old and may contain logs of an already resolved issue.

Reply:
yep glad you caught this as it was just for POC but the logic and the title both don't hold much value at this point :D

Reply:
it's inefficient and assumes the logs aren't going to be huge in size + no timestamp comparisons

the title prints out 'Velero logs for pod [node-agent]' without pointing out the pod name.